osgp/platform/osgp-adapter-domain-smartmetering/src/main/java/org/opensmartgridplatform/adapter/domain/smartmetering/application/services/MBusGatewayService.java

```diff
@@ -118,12 +119,11 @@ public void deCoupleMbusDevice(final DeviceMessageMetadata deviceMessageMetadata
 
         // If Mbus device is already decoupled, return response OK, otherwise,
         // decouple it.
-        if (this.isMbusDeviceCoupled(mbusDevice)) {
+        if (!this.isMbusDeviceCoupled(mbusDevice)) {
```

Comment:
this was a bug ?

Reply:
Bug was in the isMbusDeviceCoupled method. Now return boolean value that corresponds with the method  name.